src/memory/inventory_manager.rs

```diff
@@ -0,0 +1,101 @@
+use crate::memory::memory_context::MemoryContext;
+use crate::memory::{MemoryManager, MemoryManagerUpdate};
+use crate::state::StateContext;
+use data::prelude::all_items;
+use log::info;
+use memory::game_engine::il2cpp::unity_serializable_dictionary::*;
+use memory::memory_manager::il2cpp::UnityMemoryManager;
+use memory::process::{MemoryError, Process};
+use memory::string::ArrayWString;
+use std::hash::Hash;
+
+impl Default for MemoryManager<InventoryManagerData> {
+    fn default() -> Self {
+        let manager = Self {
+            name: "InventoryManager".to_string(),
+            data: InventoryManagerData::default(),
+            manager: UnityMemoryManager::default(),
+        };
+        info!("Memory: {} Loaded", manager.name);
+        manager
+    }
+}
+
+#[derive(Default, Debug)]
+pub struct InventoryManagerData {
+    // This references QuantityByInventoryItemReference which is a
+    // SerializableDictionary<InventoryItemReference, int> which is
+    // T1 = Guid
+    // T2 = Quantity
```

Comment:
T1 and T2 should probably be referenced somewhere (or change to the type names above instead?)